python/shotgun_model/shotgun_query_model.py

```diff
@@ -957,6 +958,11 @@ def __on_sg_data_arrived(self, sg_data):
         # pre-process data
         sg_data = self._before_data_processing(sg_data)
 
+        # take a copy of the data cache before we add any new items to it.
+        # this is so we can compare the pre updated cache against the current items in the model
+        # So we can work out if we want add the new items now, or wait until the fetchMore adds all the cached items
+        pre_update_cache = copy.copy(self._data_handler._cache)
```

Comment:
I agree with your comment that this is not ideal (accessing a protected member of the data handler). Maybe we could add a method to the data handler to call into the cache: `cached_child_uids = self._data_handler.get_child_uids(uid)`.